dvc/config.py

```diff
@@ -491,6 +495,8 @@ def set(self, section, opt, value, level=None, force=True):
             )
 
         config[section][opt] = value
+
+        self.validate(config)
```

Comment:
Ok, but now if you have an already broken config, you won't be able to fix it with `dvc config`, because you will be hit with this error. Plus merging configs will be broken too (e.g. url is in config.local but some params are in config, which will raise ConfigError now if you try to modify `config`, because there is no URL there).

Reply:
@efiop , `dvc config --unset` or `dvc remote modify --unset` won't work if you entered an invalid key.

```console
$ dvc init --no-scm
$ dvc remote add example https://example.com
$ dvc remote modify example invalid value
$ dvc version
ERROR: configuration error - extra keys not allowed
$ dvc remote modify --unset example invalid
ERROR: configuration error - extra keys not allowed
```

Here's the resulting config file.
```ini
['remote "example"']
url = https://example.com
invalid = None
```

Although, It might be a problem with `--unset`.

---

> Plus merging configs will be broken too.

You are totally right. Fixing it on the next commit.

Reply:
@mroutis Yes, it won't work with your PR, that is what I'm saying. Before your PR it did work, so if your colleague messed something up, you could unset stuff (e.g. started using some new config option from a newer dvc release). Maybe that is an ok approach, not sure.

Reply:
@efiop , what do you mean? The example that I shared with `--unset` doesn't work on master.